### DISCUSSION TITLE
writeContract vs sendTransaction

### DISCUSSION THREAD

**Discussion by Kamlesh-62** [wevm/wagmi#4116] [Q&A]
Hello,

I am working on a project where users connect with WalletConnect, choose their wallet, and pay using our token. Here is the code to pop up the wallet for confirmation of the transfer:

```
const amount = ethers.utils.parseEther(numberOfToken.toString());

// Importing the ABI
const abi = TokenABI.abi;

// Send the transaction using wagmi's useSendTransaction
const { request } = await simulateContract(customConfig, {
    abi,
    address: TestExitTokenContractAddress,
    functionName: 'transfer',
    args: [
        ReceiverAddress,
        amount
    ],
    account: address,
});
const hash = await writeContract(customConfig, request);
const receipt = await waitForTransactionReceipt(customConfig, {
    hash: hash
});
```

When the user triggers the function, they see the confirmation of the transfer on the selected wallet (see attachment).

<img width="360" alt="Screenshot 2024-07-05 at 11 36 02 AM" src="https://github.com/wevm/wagmi/assets/85969474/8b715917-cd63-445b-828c-1604d94bdb8f">

<img width="374" alt="Screenshot 2024-07-05 at 11 41 00 AM" src="https://github.com/wevm/wagmi/assets/85969474/abbacefb-d26e-4cf4-86b8-e631073c9f85">

However, when the user selects Trust Wallet, the transfer should be in "TestExitTokenContractAddress" but it is showing in BNB. How can I fix this?

Thank you

**Comment by chuksys** [comment]
Did you find a solution to this problem?
